### PR TITLE
timeout

### PR DESCRIPTION
**Pull request by zilohumberto**

### CODE REVIEW

File: backend/settings/default.py [hunk 1]
```diff
@@ -29,7 +29,8 @@
 
 # TASK MANAGER SETTINGS
 KEEP_RUNNING_TASK_MANAGER: bool = True
-MAX_TIMEOUT_SECONDS: int = 5 * 60  # 5 minutes = 300 seconds
+MAX_TIMEOUT_SECONDS: int = int(os.getenv("MAX_TIMEOUT_SECONDS") or 5 * 60)  # 5 minutes = 300 seconds
+NUMBER_OF_ACTIVE_TASK_PER_MANAGER = int(os.getenv("NUMBER_OF_ACTIVE_TASK_PER_MANAGER"))
```
Comment (anish749) [inline backend/settings/default.py:33]: in our first impl, we need to have a limit on the number of active tasks per manager.. have you already implemented this?

File: backend/application/services/queue/consumer.py [hunk 2]
```diff
@@ -6,85 +7,89 @@
...
+    if task_id in listeners:
+        del listeners[task_id]
+    """
+
+from application.services.tasks_definition.wait_5s import wait_5s
+
 async def consumer():
+    logging.info("LOL!!!!")
+    try:
+        cm = await TASK_MAP["wait_5s"](task_id="LOL!")
```
Comment (anish749) [inline backend/application/services/queue/consumer.py:42]: what is this going on here? what are you trying to test?

File: backend/application/services/queue/consumer.py [hunk 3]
```diff
@@ -6,85 +7,89 @@
...
+                    f"task getting {task_record.name} [{task_record.status.name}]-> {task_id}"
                 )
 
-                # check if task is done with his purpose of live, it not can not be deleted
-                queue_message_done = False
                 if task_record.name in TASK_MAP:
-                    if task_kind == MessageKind.task:
-                        repository.mark_running(obj=task_record)
+                    # repository.mark_running(obj=task_record)
+                    async with asyncio.timeout(0.1):
```
Comment (anish749) [inline backend/application/services/queue/consumer.py:72]: awaiting with a timeout on a per task inside a loop would effectively slow down the loop.

**Comment by anish749** [comment]
Here is an example of using asyncio for something like timeouts:

```python
import asyncio


async def summer(a, b):
    print("summer started")
    await asyncio.sleep(1)
    print("summer done")
    return a + b


async def multiply(a, b):
    print("multiply started")
    await asyncio.sleep(1)
    print("multiply done")
    return a * b


summer_task = summer(1, 2)
multiply_task = multiply(1, 2)

async def with_timeout(coroutine, timeout):
    try:
        result = await asyncio.wait_for(coroutine, timeout=timeout)
        return result
    except asyncio.TimeoutError:
        print("timeout")


async def run_all_tasks():
    awaited = await asyncio.gather(
        with_timeout(summer_task, timeout=0.5),
        with_timeout(multiply_task, timeout=1.5),
    )


asyncio.run(run_all_tasks())
```

File: backend/settings/default.py (hunk 1 above)
Comment (zilohumberto) [reply backend/settings/default.py:33]: no yet

File: backend/application/services/queue/consumer.py (hunk 2 above)
Comment (zilohumberto) [reply backend/application/services/queue/consumer.py:42]: yes

**Comment by zilohumberto** [comment]
> Here is an example of using asyncio for something like timeouts:
> 
> ```python
> import asyncio
> ...
> 
> asyncio.run(run_all_tasks())
> ```

This sample is ok, and is close what I saw on docs however I do not see how it can I plug it into my code. 
Do I'm following a wrong approach? or am I missing something? 

My thread start on 
> asyncio.run(consumer())

how can I run a gather for an unknown number of task, or even more, or when trigger the task that is on demand ( depends on the queue)

**Comment by anish749** [comment]
> > Here is an example of using asyncio for something like timeouts:
> > ```python
> > import asyncio
> > ...
> > 
> > asyncio.run(run_all_tasks())
> > ```
> 
> This sample is ok, and is close what I saw on docs however I do not see how it can I plug it into my code. Do I'm following a wrong approach? or am I missing something?
> 
> My thread start on
> 
> > asyncio.run(consumer())
> 
> how can I run a gather for an unknown number of task, or even more, or when trigger the task that is on demand ( depends on the queue)

that would be a different task.

so we run 
```python

def main():
  loop.create_task(consume_messages())
  loop.run_forever()

async def consume_messages():
  with True:
    await queue.get_messages(...)
   await asyncio.gather(...)
```

File: backend/application/sample/micro_consumer_timeout.py
Comment (zilohumberto) [inline backend/application/sample/micro_consumer_timeout.py:1]: I think this sample sounds more like what we need and expected. @anish749 what do you think?

Comment (anish749) [reply backend/application/sample/micro_consumer_timeout.py:1]: yes, this looks good to me. you are on the right track.

Comment (zilohumberto) [reply backend/application/sample/micro_consumer_timeout.py:1]: I have other question.

async def consume_messages():
    while True:
        task = await produce_fake_messages() # we are simulating here the queue 
        if task:
            await asyncio.gather(with_timeout(task, timeout=5))  # we wait here until the task comple, if there are new messages we ignored until the currently task it's done, that is not exactly what I expected.
           # I'm searching for a way to run asyncio.gather without awaiting it and still handling the timeouts.  
 @anish749 do you see my point?

Comment (anish749) [reply backend/application/sample/micro_consumer_timeout.py:1]: `I'm searching for a way to run asyncio.gather without awaiting it and still handling the timeouts.`
to achieve this you just remove the `await` keyword.

eg: 
```python
tasks = asyncio.gather(...)

# later on.
await tasks 
```

**Pull request closed without merging**